### PR TITLE
abinit: add version 9.10.5

### PR DESCRIPTION
**Pull request by downloadico**
Add new version 9.10.5.
Also, set the FC variable to the MPI Fortran compiler and also set the F90 variable to the same compiler for versions 9.8 and up.  FC needs to be set because the configure script still uses FC.

### CODE REVIEW

**Review by tldahlgren — APPROVED** [review]
Confirmed the version sha256.

**Comment by tldahlgren** [comment]
@downloadico Do you want me to wait for Todd to review before merging this PR?

**Comment by downloadico** [comment]
> @downloadico Do you want me to wait for Todd to review before merging this PR?

@tldahlgren No, I got a little "click happy" :)

**Pull request merged**
